### ISSUE TITLE
[WordPress] Setting up API access to WordPress on AWS EC2 instance

### ISSUE THREAD

**Issue by agomezmoron** [bitnami/vms#18]
### Platform

AWS

### bndiagnostic ID [know more about bndiagnostic ID](https://docs.bitnami.com/general/how-to/understand-bndiagnostic/)

a1ad2025-43cb-ba0a-4de3-3beda30f27b4

### bndiagnostic was not useful. Could you please tell us why?

I didn't find any solution in the documentation.

### Describe your issue as much as you can

(migrated from community.bitnami.com)

Hello all,

I have WordPress installed on an AWS EC2 instance (t2.micro).
I have the default admin user ‘user’, and I have created one other user who is an author.
Everything worked fine initially, in terms of accessing the site, creating posts and performing admin duties etc.

I wanted to make use of the WordPress API. I’ve been doing some experiments with POSTMAN. I’ve been able to GET posts, comments, users etc OK but am unable to POST using the API. I get back this response:

{“code”:”rest_cannot_create”,”message”:”Sorry, you are not allowed to create posts as this user.”,”data”:{“status”:401}}

I’ve installed and activated the plugin referenced here:
[https://developer.wordpress.org/rest-api/using-the-rest-api/authentication/ 154](https://developer.wordpress.org/rest-api/using-the-rest-api/authentication/)

The plugin itself is here:
[https://github.com/WP-API/Basic-Auth 12](https://github.com/WP-API/Basic-Auth)

I’m trying to POST using Basic Authenticiation and have tried with the credentials for each user, but I’m still getting the same response as above.

A moderator on the WordPress forums suggested I take a look at this:
[https://github.com/WP-API/node-wpapi/issues/349 10](https://github.com/WP-API/node-wpapi/issues/349)

The post on 31 March in that thread suggests installing and activating the ‘Application Passwords’ plugin, which I’ve now done. It also says to then follow the instructions here: [https://github.com/georgestephanis/application-passwords/wiki/Basic-Authorization-Header—-Missing 102](https://github.com/georgestephanis/application-passwords/wiki/Basic-Authorization-Header%E2%80%94-Missing)

Those instructions suggest an edit to the ‘htaccess’ file.

So I’ve intalled the ‘wp htaccess control’ plugin: [https://wordpress.org/plugins/wp-htaccess-control/ 70](https://wordpress.org/plugins/wp-htaccess-control/)
and edited the htaccess file as advised.
I’ve also created some application passwords for the two users.

I’m still getting the same issue when I try to POST to WordPress from POSTMAN.
Worse, now when someone visits my site (just through their browser) they are prompted to enter a username and password. My user credentials do NOT grant access. However, I am still able to log on to the Wordpress admin console, and I am still able to GET posts, comments, user details etc using the API via Postman. So it seems that somewhere along the way I have inadvertantly made passwords required just to access the site via a browser.

One other point: When I Putty onto my EC2 instance, I cannot see the htaccess file in any directory. Although I can see its content from within my admin console using the ‘wp htaccess control’ plugin.

In short, I’m bewildered, and in need of some advice. What is the ‘best practice’ way to fully enable use of the WordPress API, running on an EC2 instance?
I appreciate this might not be the ideal place to post about these issues, but I’ve not yet got definitive advice from either the WordPress forums or the AWS forums, so I’m hoping I might find some helpful across-the-board expertise here :slight_smile:

Regards
Jamie.
